Telink b95 new old patches

haiwentelink opened the pull request:
update the ske part , and update the flash.c and flash.h for flash_read_mid interface.

pulin1103 commented on tlsr9/drivers/B95/usbhw.h:
It will disable the enabled endpoints.

pulin1103 commented on tlsr9/drivers/B95/usbhw.h:
```diff
@@ -337,7 +337,7 @@ static inline void usbhw_set_eps_en(usb_ep_en_e ep)
     else
     {
-        reg_usb_edp_en |= ep;
+        reg_usb_edp_en = ep;
```
This will disable the enabled endpoints.

haiwentelink replied:
follow the driver in new ble sdk

pulin1103 commented on tlsr9/drivers/B95/usbhw.c:
```diff
@@ -31,8 +31,6 @@
 void usbhw_init(void)
 {
     pm_set_dig_module_power_switch(DIG_MODULE_USB, PM_POWER_UP);
-    pm_wait_dig_module_power_stable();
```
Why is this function commented?

haiwentelink replied:
in the new ble sdk , the func  pm_wait_dig_module_power_stable is disapperared .

s07641069 commented on zephyr/module.yml:
```diff
@@ -22,10 +22,10 @@ blobs:
+    url: https://github.com/telink-semi/zephyr_hal_telink_b91_ble_lib/raw/17178655138c95075056dc433562e58b9c08052a/liblt_9258_zephyr.a
     description: "Binary libraries supporting Telink B95 series BLE subsystems"
-    doc-url: https://github.com/telink-semi/zephyr_hal_telink_b91_ble_lib/blob/49bcada9aae1eaae83aee4aa437244403aaf2162/README.md
+    doc-url: https://github.com/telink-semi/zephyr_hal_telink_b91_ble_lib/blob/17178655138c95075056dc433562e58b9c08052a/README.md
```
change hash after merge of https://github.com/telink-semi/zephyr_hal_telink_b91_ble_lib/pull/15

s07641069 commented on tlsr9/ble/vendor/controller/b9x_bt.c:
```diff
@@ -28,7 +28,11 @@
 #include "b9x_bt.h"
 #include "b9x_bt_init.h"
 #include "compiler.h"
+#if CONFIG_SOC_RISCV_TELINK_B91 || CONFIG_SOC_RISCV_TELINK_B92
 #include "plic.h"
+#elif CONFIG_SOC_RISCV_TELINK_B95
+#include "lib/include/plic.h"
+#endif
```
better to add "lib/include/" into CMake include list and revert this change

haiwentelink replied:
will add the lib/include into cmake list .

s07641069 commented on tlsr9/ble/common/utility.h:
```diff
@@ -101,7 +101,7 @@
 #define foreach_arr(i, arr) 	for(int i = 0; i < ARRAY_SIZE(arr); ++i)
 
-#define ARRAY_SIZE(a) 			(sizeof(a) / sizeof(*a))
+// #define ARRAY_SIZE(a) 			(sizeof(a) / sizeof(*a))
```
don't leave commented code.
remove if not needed.

haiwentelink replied:
will delete

s07641069 commented on tlsr9/common/b9x_rf_power.c:
```diff
@@ -17,7 +17,11 @@
+#if CONFIG_SOC_RISCV_TELINK_B91 || CONFIG_SOC_RISCV_TELINK_B92
+ #include <rf.h>
+#elif CONFIG_SOC_RISCV_TELINK_B95
+#include <lib/include/rf/rf_common.h>
```
after adding "lib/include/" into CMake include list use "rf/rf_common.h"
as well to avoid lots of #if - you can put "rf/rf_common.h" into "rf.h" or something like this

haiwentelink replied:
will add the lib/include /rf into cmake list .

s07641069 commented on tlsr9/crypto/mbedtls/internal/compatibility/aes_alt.c:
```diff
@@ -54,8 +54,8 @@
 #if CONFIG_SOC_RISCV_TELINK_B95
-#include "ske.h"
-#include "ske_portable.h"
+#include "lib/include/ske/ske.h"
+#include "lib/include/ske/ske_portable.h"
```
As well "lib/include" can be removed from path

haiwentelink replied:
will add the lib/include /ske into cmake list .

s07641069 commented on tlsr9/crypto/mbedtls/internal/compatibility/ecp_alt.c:
```diff
@@ -83,8 +83,16 @@
 #include <pke.h>
-#if (CONFIG_SOC_RISCV_TELINK_B92 || CONFIG_SOC_RISCV_TELINK_B95)
+#elif CONFIG_SOC_RISCV_TELINK_B95
+#include <lib/include/pke/pke.h>
```
same  "lib/include" can be removed from path and after #if can be reverted (simplified)

haiwentelink replied:
will add the lib/include /pke into cmake list .

s07641069 commented on tlsr9/crypto/mbedtls/internal/compatibility/ecp_alt.c:
```diff
@@ -83,8 +83,16 @@
+#elif CONFIG_SOC_RISCV_TELINK_B95
+#include <lib/include/pke/pke.h>
+
+#include <ext_driver/driver_lib/ext_lib.h>
+#include <ext_driver/ext_misc.h>
+#endif
```
why we can't use same path for all SoCs?

haiwentelink replied:
for the driver file changes , and we don't want to to adjust the drive file .

s07641069 replied:
ok

s07641069 commented on tlsr9/crypto/mbedtls/internal/ecp_alt_b9x_backend.c:
```diff
@@ -45,9 +45,16 @@
+#if CONFIG_SOC_RISCV_TELINK_B91 || CONFIG_SOC_RISCV_TELINK_B92
 #include <pke.h>
+#elif CONFIG_SOC_RISCV_TELINK_B95
+#include <lib/include/pke/pke.h>
```
same here
clear for all changes include paths to use same includes and avoid lots of #if

andriy-bilynskyy commented:
PR seems fine. But plese squash your commits. There is no any reasons to keep middle sources version where builds are not tested.

haiwentelink commented:
Hi andriy , for the middle changes , we tested every version , i don't understand , so can you point out which commit need to squash , from xxx to xxx ? I guess from b61e2250d056abba92d0a7dc8fd78d79264de2fd to 1568eebcf2fbe382f70f5b683afd33ac94cbd216 ? For some commit should not be squash which we need to do patch for next version .

s07641069 commented:
@haiwentelink 
merge https://github.com/telink-semi/zephyr_hal_telink_b91_ble_lib/pull/15 (approved) and after update in this PR hash of lib (do again force push to have 1 commit)

haiwentelink commented:
already update the hash of the of libs .

haiwentelink committed:
hal: B95: clean code and fix the sys_reboot_lib hangs in bootloader.

- fix sys_reboot_lib in usbhw.c.
- clean code for the file path changed .
- set the new lib for B95 .
- special setting for ext_pm.h to avoid compile warnings .
- update ble lib for CI .
- fix ske bugs .
- update rf reset API .

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>
